### DISCUSSION TITLE
Planning buffer default size.

### DISCUSSION THREAD

**Discussion by phil-barrett** [grblHAL/core#402] [General]
The default planning buffer size is 35.  I believe this dates from the original 8-bit Grbl and made sense then because of the tiny amount of RAM on the ATMega.  Any thought to increasing the default to something larger?  Say 128?  Any reason not to?  Maybe target the smallest processor RAM out there - STM32F401?

I was looking at some benchmarks for laser machines and the grblHAL ones were run with the default planning buffer size.  When I pointed this out, the person reran a few tests and got significant improvement.

**Comment by terjeio** [comment]
I'll increase the default for all but the STM32F103 which is the one with the smallest amount of RAM available (20K).
Note that if the default (or $398) is set to a value too high for the available RAM it will be automatically decreased until the planner buffer fits.

**Reply by phil-barrett** [reply]
That's great.  And, yeah, I was thinking of the F103 when wrote F401.
